frontend/src/ClientDataForm.js

```diff
@@ -116,19 +122,57 @@ const ClientDataForm = () => {
     </div>
   );
 
+  if (loading) {
+    return (
+      <>
+        <div className="container d-flex justify-content-center align-items-center min-vh-100">
+          <div className="spinner-grow text-primary" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-secondary" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-success" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-danger" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-warning" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-info" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-dark" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+        </div>
+      </>
+    );
+  }
+
   return (
     <>
       <div className="container">
         <div className="row mb-2 border-bottom">
           <h1>Client Form</h1>
         </div>
         <form method="post" onSubmit={log_information}>
-          {questions.map((question, index) => renderQuestion(question, index))}
+          {questions
+            .filter((question) => question.language_fk_id == language)
```

Comment:
I think it should be `===` rather than `==` here for equal operation.

Reply:
=== breaks it and == works for some reason, even though JS throws a warning that it expects ===

Reply:
This is because `language` is a string but `question.language_fk_id` is an int/number so we'll need to convert `language` to int/number first before we can use the `===` for comparison which checks for type as well. 

I think you can do `question.language_fk_id === Number(language)` here and that should work.